pages/game.tsx

```diff
@@ -1,22 +1,127 @@
-import React from "react";
-import Link from "next/link";
-import { Container } from "react-bootstrap";
+import Board from "../components/Board";
+import Keyboard from "../components/Keyboard";
+import { createContext, useState, Dispatch, SetStateAction } from "react";
+import { boardDefault, words } from "../components/words";
+import GameOver from "../components/GameOver";
+interface IContext {
+  board: string[][];
+  setBoard: (board: string[][]) => void;
+  currAttempt: {
+    rowIndex: number;
+    letterIndex: number;
+  };
+  setCurrAttempt: (currAttempt: {
+    rowIndex: number;
+    letterIndex: number;
+  }) => void;
+  onEnter: () => void;
+  onDelete: () => void;
+  onLetter: (value: string) => void;
+  correctWord: string;
+  disabledLetters: string[];
+  setDisabledLetters: Dispatch<SetStateAction<string[]>>;
+  correctLetters: string[];
+  setCorrectLetters: Dispatch<SetStateAction<string[]>>;
+  closeLetters: string[];
+  setCloseLetters: Dispatch<SetStateAction<string[]>>;
+  gameOver: { gameOver: boolean; guessedWord: boolean };
+  setGameOver: (gameOver: { gameOver: boolean; guessedWord: boolean }) => void;
+}
+
+export const GameContext = createContext<Partial<IContext>>({});
+
+function game() {
+  const [board, setBoard] = useState(boardDefault);
+  const [currAttempt, setCurrAttempt] = useState({
+    rowIndex: 0,
+    letterIndex: 0,
+  });
+  const correctWord = "RIGHT";
+  const [disabledLetters, setDisabledLetters] = useState<string[]>([]);
+  const [correctLetters, setCorrectLetters] = useState<string[]>([]);
+  const [closeLetters, setCloseLetters] = useState<string[]>([]);
+  const [gameOver, setGameOver] = useState({
+    gameOver: false,
+    guessedWord: false,
+  });
+  const onEnter = () => {
+    // do not allow entering if there are not enough letters
+    if (currAttempt.letterIndex != 5) return;
+
+    const entry = board[currAttempt.rowIndex].join("").toLowerCase();
+
+    // check if we should move onto the next row -> only if the word is a valid word
+    if (words.includes(entry)) {
+      setCurrAttempt({
+        rowIndex: currAttempt.rowIndex + 1,
+        letterIndex: 0,
+      });
+    }
+
+    // check if the entry is correct. if so, end the game
+    if (correctWord.toLowerCase() === entry) {
+      setGameOver({ gameOver: true, guessedWord: true });
+      return;
+    }
+
+    // check if user reached maximum attempts
+    if (currAttempt.rowIndex == 5) {
+      setGameOver({ gameOver: true, guessedWord: false });
+    }
+  };
+  const onDelete = () => {
+    if (currAttempt.letterIndex == 0) return;
```

Comment:
Generally you should use `===` instead of `==` and `!==` instead of `!=` since this way is more safe, especially when comparing with values like 0, which could be considered "equal" to values like false or empty string.